README.md
point-q
=======

Point-queue simulator in Python3


Instructions for an implementation

 
1)  Fill the «  Dsu_1.py »  file (file located in folder cc)

2) In the folder « Control_Param_Files »  (folder in folder cc) indicate the file with the control parameters and the file indicating the the nice id  and the type of the employed control.

More precisely in the folder « Control_Param_Files » two files should  exist:

i) « fi_node_id_ctrl_type_category.txt »  

A priori each node may have a different type of control. 

File « fi_node_id_ctrl_type_category.txt »  indicates  which type of control  should be applied at each intersection node. 

Each  line of this file corresponds to a specific node.

For each line, four columns correspond.

The first column indicated the node id. 

The second column indicates the type of the control.

The control types available at this moment are:

0:"type_control_red_clear",
1:"type_control_FT", 
2:"type_control_FT_Offset", 
3:"type_control_MP",
10:"type_control_FA_no_red_clear",
11:"type_control_FA_Max_Green",\
12:"type_control_FA_with_red_clear",
13:"type_control_MP_Practical",
14:"type_control_MP_without_red_clear",\
15:"type_control_MP_Practical_without_rec_clear"

remarque: if there are missing numbers it is because there exist controls which are not currently employed (mixed, MP with boundary controls, MP without negative pressure etc).

The 3rd column may take two possible values:

a) « without_sensor_requirement » if the employed control is updated at predefined times (which do not depend from the current values of flows or queues. There are the FT control, FT with offsets, all versions of MP)

b) `` with_sensor_requirement ‘’ if the employed control is updated according to the current values of flows or queues (e.g. FA, FA with max green etc.).

The fourth column indicates if the turn ratios are going to be estimated (value 1) or not (value 0) during the implementation. This make sense for controls   selecting a stage  with use of the saturation flows (for exemple, MP, FA etc).


ii) A file indicating the control parameters

If  it is FT, FT with offsets or FA (not working yet….) only one file is required.
If a version of MP is employes two files will be required. 

More precisely.

a) For a FT this file  is called «File_FT_Control_Alg_Param.txt» 
b) For FT with offsets this file is called « File_FT_Offset_Control_Alg_Param.txt »
c) For MP these file are  called ``File_MP_Control_Alg_Param_res.txt ‘’ and ``File_MP_Qvalues_phases.txt »
d) for MP without red clearance, these files are called ``File_MP_without_rc_Control_Alg_Param.txt’’ and NOT YET completed
e) For MP Practical these files should be called ``File_MP_Practical_Control_Alg_Param.txt’’  and NOT YET completed
f) for FA with red clearance this file should be called ``File_FA_with_red_clear_Control_Alg_Param.txt »
g)for FA with red clearance this file should be called "File_FA_no_red_clear_Control_Alg_Param.txt"
h) for FA with max green, this file should be called "File_FA_MAX_GREEN_Control_Alg_Param.txt"


Here after is a short explanation of each file, for which there is also an exemple for the network of 15 nodes in the folder Control_Param_Files

a1) File_FT_Control_Alg_Param.txt

each line corresponds to a node.`
The first line of this file explains  each column

For any other line:

- the 1st column indicates the node id

- the second column indicates the stage id. A strictly positive number if the stage is not red clearance, 0 otherwise.
Stages are indicated in the file ``fi_stages_each_sign_inters.txt » included int he network folder.

- the 3rd column indicates the actuation duration of the stage (in secs)

- the 4th column indicates the cycle duration (this value however is not explicitly employed by the simulator).


b1)  ‘’File_FT_Offset_Control_Alg_Param.txt’’
The The first line of this file explains  each column.

The rest of the  lines.

2nd line indicates the master node id

3rd line indicates a sequence of (node id, value offset).

All other lines:

1st column: node id 

- the second column indicates the stage id. A strictly positive number if the stage is not red clearance, 0 otherwise.
Stages are indicated in the file ``fi_stages_each_sign_inters.txt » included int he network folder.

- the 3rd column indicates the actuation duration of the stage (in secs)

- the 4th column indicates the cycle duration (this value however is not explicitly employed by the simulator).



c1) File_MP_Control_Alg_Param_res.txt

The first line is comments

For all other lines: 

1st column: node id
2nd column : stage actuation duration
3rd column: duration of red clearance
 
All the other columns of the same line should be a sequence of (2nd column, 3rd column). 

c2) `File_MP_Qvalues_phases.txt »

First line should be a comment

next lines:

1st column:  id input link forming the phase
2nd column:  id ioutput link forming the phase
3rd column; Q(l,m) value


d1) ``File_MP_without_rc_Control_Alg_Param.txt’’

1st line s comments
Next lines:

1st column=id node
2nd column : control actuation duration

d2) NOT  YET





e1) ``File_MP_Practical_Control_Alg_Param.txt’’ 


1st line s comments
Next lines:

1st column: id node
2nd column stage actuation duration
3rd column: duration red clearance
4th column: parameter value for switching

e2) Not yet


f1) ``File_FA_with_red_clear_Control_Alg_Param.txt »
1st line s comments
Next lines:


1st column: id node
2nd column: max admissible queue size
3rd column: red clearance duration
4th column t start first control


g1) "File_FA_no_red_clear_Control_Alg_Param.txt"

1st line s comments
Next lines:
 
1st column: id node
2nd column: max admissible queue size
3rd column t start first control

h1) "File_FA_MAX_GREEN_Control_Alg_Param.txt"

1st line s comments
Next lines:
1st column: id node
2nd column: max admissible queue size
3rd column: red clearance duration
4th column t start first control
5th column: max green duration

3) Execute file SImulation.py

On a terminal or something equivalent (depending upon the operating system of the computer) you  move into folder  cc
 (e.g.. if you have named the simulation folder sim and you place it on the desktop , with the cd command you type

i) cd Desktop/sim/sim_1/cc and then you type enter
SO now you are in the folder cc where the python files are and also the file Simulation.py to execute (be careful, there is another file called Cl_Simulation.py. But what you have to execute  is the one called  Simulation and not Cl_Simulation.py).
ii) type python3.x Simulation.py  or time python3.x Simulation if you wish the time consumed by  your system to run the sim. 

python3.x stands for the python version you have. 3 indicates that you  can run the « .Q » with any version of python 3.

for exemple if you have python3.4 you can type:

python3.4 Simulation.py and then enter. 

The simulation starts. 

If you run the simulator with the Dsu_1.py file as I give it to you and without modifying anything in folder Control_Param_Files, you will run a new sim of 6000 secs, for a stabilising FT control. 


————————————

When a series of one or more simulations is implemented  a folder named Series_Sim_date is created.
Inside this folders there are folders named Fres_date corresponding to each simulation you run. 
Inside each folder Fres_date there are written many files employed by the current simulation (so as you will remember or verify the sim characteristics for this run).
Amongst these files there is a file named ``file_recording_event_db.txt »  with the recordings of the simulation.
This is the file I need to explain you so as to create your own statistical analysis. 





CHANGELOG.txt
CHANGELOG

**********************************************************************************************************************************************************************************
06/07/2014 01:36 am
**********************************************************************************************************************************************************************************
	- Cl_Ev_veh_appearance.py:

	Uncomment the code that allowed to update the number of vehicules in the entry link

	- Dsu_1.py

	Changed option to finite link capacity: allows to see the link evolution in Stat Analysis

	- Cl_Ev_end_veh_departure_from_que.py

	Allowed to update the number of vehicule in the links even if it's an entry or an output link

	- Cl_Stat_Analysis_new.py

	Allowed to write the txt file of the link evolution even if it's an output link

**********************************************************************************************************************************************************************************
07/07/2014 10:46 pm
**********************************************************************************************************************************************************************************
	- Cl_Simulation.py:

	Possibility to stop the simulation every 5sec to exchange data with the CTM simulator

**********************************************************************************************************************************************************************************
17/07/2014 02:27 am
**********************************************************************************************************************************************************************************
	- Possibility to choose in Dsu_1.py if we want to connect the point-q-simulator with CTM simulator (val_ctm_connect)
	- Hungtington colorado configuration files changed (mistakes corrected)



sim_1/cc/Dsu_1.py
#import File_Sim_Name_Module_Files
import Cl_Decisions
import Cl_Control_Actuate

#*****************************************************************************************************************************************************************************************
#the name of the folder where the netowrk files are registered
#val_name_folder_network_files="../SMALL_NETWS_2/RÉSEAUX/SMALL_DATA_2NDS"
#val_name_folder_network_files="../SMALL_DATA_NG_4NDS"
#val_name_folder_network_files="../SMALL_DATA_INTERS_2NDS_LK_CAP_PRIOR_PHASES"
#val_name_folder_network_files="../SMALL_NETWS_2/SMALL_round_about_jennie"
#val_name_folder_network_files="../SMALL_NETWS_2/SMALL_DATA_2NDS"
#val_name_folder_network_files="../SMALL_NETWS_2/SMALL_NETWORK_4NDS"
#val_name_folder_network_files="../SMALL_NETWS_2/SMALL_DATA_9NDS"
#val_name_folder_network_files="../SMALL_NETWS_2/SMALL_DATA_test"
#val_name_folder_network_files="../SMALL_NETWS_2/SMALL_DATA_2NDS_FIG_7"
#val_name_folder_network_files="../SMALL_NETWS_2/SMALL_DATA_2NDS_jennie"
#val_name_folder_network_files="../SMALL_NETWS_2/SMALL_round_about_jennie"
#val_name_folder_network_files="../SMALL_NETWS_2/SMALL_DATA_2NDS_FIG_1"
#val_name_folder_network_files="../SMALL_NETWS/reseaux_2/SMALL_DATA_SD"
#val_name_folder_network_files="../SMALL_NETWS/reseaux_3/SMALL_DATA_2NDS_FIG_7_papier_2"
#val_name_folder_network_files="../SMALL_NETWS/reseaux_3/SMALL_DATA_SD_sans_OD"
#val_name_folder_network_files="../SMALL_NETWS/reseaux_3/SMALL_DATA_SD_avec_OD"
val_name_folder_network_files="../SMALL_NETWS/huntington_colorado"
#val_name_folder_network_files="../SMALL_NETWS/reseaux_3/SMALL_DATA_reseau_15_nds"
#*****************************************************************************************************************************************************************************************

#variable indicating if we will  creat the text record file (value=1 the text file will be created,  0 otherwise)
# (where we write in a text file  the realisation of each event).if the value of this variable is  different from 0,1, there will be problems in the
#implementation
#creation_text_file_recording_sim_events_text=1
#creation_text_file_recording_sim_events_text=0

#*****************************************************************************************************************************************************************************************
#variable indicating if messages will be print on the terminal during the simulation (this variable does not concern the error messages, these ones
#will alwyas be printed). If its value=1  messages will be printed, 0 otherwise).if the value of this variable is  different from 0,1, there will be problems in the
#implementation
#print_messages_on_terminal=0
print_messages_on_terminal=1
#*****************************************************************************************************************************************************************************************
#variable  indicating the precision on  some rounds that we will use (depending on the t_unit)
val_precision_round_for_defin_time=1 
#*****************************************************************************************************************************************************************************************
#variable indicating the round precision on computations pressure etc.  related to control algos
val_precision_round=2
#*****************************************************************************************************************************************************************************************
#variable indicating the time unit considered
#val_t_unit=1
val_t_unit=0.1

#*****************************************************************************************************************************************************************************************
#variable indicating if we want to connect point-q simulator with the CTM simulator (1 -> yes, 0 -> no)
#val_ctm_connect = 1
val_ctm_connect = 0

#*****************************************************************************************************************************************************************************************
#variable indicating the path of the folder in which state infos exchanged with CTM will be
val_fol_ctm_connect = 'exchange_zone'

#*****************************************************************************************************************************************************************************************

#variable indicating whether a stochastic or deterministic (1, 0)  demand is been employed
val_indicating_stoch_demand=0
#val_indicating_stoch_demand=1

#*****************************************************************************************************************************************************************************************
#variable indicating if we wish to create a new demand or not. If its value = 1, new vehicles will be generated to each entry link, 
#if  its value = 0, a previously generated demand will be employed,
#if  its value = -1, a given (generated) demand will be employed,
creation_new_demand=1
#creation_new_demand=0
#creation_new_demand=-1
#*****************************************************************************************************************************************************************************************

#variable indicating the path folder containing the results of the previously generated demand, (variable to be filled if 'creation_new_demand'=0)
#path_prev_generated_veh_demand="./FRes-Thu-26-Jul-2012_13-10-29/veh_appearance_after_end_sim_in_event_list"
#path_prev_generated_veh_demand="./SIMS_2ND_MP_D2=FT/FRes-Fri-19-Oct-2012_17-19-25"
#path_prev_generated_veh_demand="./FRes-Thu-18-Oct-2012_00-31-32"
#path_prev_generated_veh_demand="FRes-Fri-16-Nov-2012_21-45-55"
#path_prev_generated_veh_demand="./Series_Sim-Wed-07-Aug-2013_20-24-54/FRes-Wed-07-Aug-2013_20-24-54"
#path_prev_generated_veh_demand="./S_RP_8dec_2e_implem_3h/FRes-Thu-19-Sep-2013_10-28-28"
#path_prev_generated_veh_demand="./S_FA_MAX_GREEN_res_2nds/FRes-Fri-03-Jan-2014_11-58-35"
path_prev_generated_veh_demand="./Series_Sim-Fri-02-May-2014_19-18-23/FRes-Fri-02-May-2014_19-18-23"
#*****************************************************************************************************************************************************************************************
#variable indicating the reason of employing a previously generated demand (variable to be filled if 'creation_new_demand'=0)
#if Cl_Decisions.TYPE_REASON_EMPLOYING_PREVIOUS_DEMAND["CTRL_EVAL"] is selected  it is because a signal control policy will be evaluated
# If  Cl_Decisions.TYPE_REASON_EMPLOYING_PREVIOUS_DEMAND["ROUT_ALGO_EVAL"] is selected  it is because a routing policy will be evaluated
#in which case, val_type_veh_final_dest=Cl_Decisions.TYPE_VEHICLE_FINAL_DESTINATION_AND_PATH["initially_defined_and_path_dyn_constructed"]

reason_employing_previous_demand=Cl_Decisions.TYPE_REASON_EMPLOYING_PREVIOUS_DEMAND["CTRL_EVAL"]
#reason_employing_previous_demand=Cl_Decisions.TYPE_REASON_EMPLOYING_PREVIOUS_DEMAND["ROUT_ALGO_EVAL"]
#*****************************************************************************************************************************************************************************************
#variable indicating if we wish to do a new simulation  or not. If its value = 1 a new simulation will be realised, if its value =0 a previous one 
#will be continued.if the value of this variable is  different from 0,1, there will be problems in the
#implementation
new_simulation=1
#new_simulation=0
#*****************************************************************************************************************************************************************************************
#the path folder where the  files with the final state of a simulation are registered, utilised when we wish to continue a previously made simulation
#this variable will be utilised when new_simulation=0, (variable to be filled if 'new_simulation'=0)
#path_name_folder_results_sim_to_be_continued="./FRes-Tue-23-Oct-2012_22-12-26"
#path_name_folder_results_sim_to_be_continued="./SIMS_2ND_FT_D2_L2/FRes-Fri-19-Oct-2012_11-56-46"
#path_name_folder_results_sim_to_be_continued="./Series_Sim-Fri-16-Aug-2013_22-42-11/FRes-Fri-16-Aug-2013_22-42-11"
path_name_folder_results_sim_to_be_continued="./Series_Sim-Fri-02-May-2014_19-18-23/FRes-Fri-02-May-2014_19-18-23"
#*****************************************************************************************************************************************************************************************
#the time at which we wish to start a new simulation.
#this variable will be utilised when new_simulation=1
t_start_new_simulation=0

#*****************************************************************************************************************************************************************************************
#the time during which we wish to simulate
#t_simulation_duration=10
#t_simulation_duration=50
#t_simulation_duration=70
#t_simulation_duration=100
#t_simulation_duration=150
#t_simulation_duration=200
#t_simulation_duration=270
#t_simulation_duration=250
#t_simulation_duration=300
#t_simulation_duration=320
#t_simulation_duration=420
#t_simulation_duration=500
#25 cycles
#t_simulation_duration=700
#t_simulation_duration=900
#t_simulation_duration=930
#t_simulation_duration=1000
#each cycle is 24 sec, we sim for 50 cycles
t_simulation_duration=1200
#t_simulation_duration=1280
#t_simulation_duration=1800
#t_simulation_duration=1802
#t_simulation_duration=2370
#t_simulation_duration=2400
#t_simulation_duration=2703
#t_simulation_duration=3000
#t_simulation_duration=3600
#t_simulation_duration=5000
#t_simulation_duration=6000
#t_simulation_duration=10000
#t_simulation_duration=10800
#t_simulation_duration=21600
#t_simulation_duration=7200
#t_simulation_duration=36000
#t_simulation_duration=28800

#*****************************************************************************************************************************************************************************************
#the min margin at which the new network control object (for the next cycle) mut be returned.
#It T is the cycle duration, the new event end decision network control must be realised at T-dt, not later than that
#margin_dt=1
margin_dt=2
#margin_dt=4
#margin_dt=9
#margin_dt=10
#margin_dt=15
#margin_dt=20
#margin_dt=0

#*****************************************************************************************************************************************************************************************
#the cycle duration in secs, employed inthe stat analysis, when calc nb veh ar/dep per cycle, 
#il n'est pas ds Cl_Data
cycle_duration=60
#cycle_duration=90
#cycle_duration=74
#cycle_duration=62
#*****************************************************************************************************************************************************************************************
#variable indicating if  internal links have (not) finite capacity (it is employed by the stat analysis. Thsi value is also indicated in the network data model)
#also employed by the event end veh depart
#val_finite_capacity_internal_links=1 #if internal  lks have finite capacity, 0 otherwise
val_finite_capacity_internal_links=0
#val_finite_capacity_internal_links=1
#*****************************************************************************************************************************************************************************************

#the min hold time of a vehicle in a queue
#val_min_hold_t_veh_in_que=1
val_min_hold_t_veh_in_que=0.1
#val_min_hold_t_veh_in_que=0

#*****************************************************************************************************************************************************************************************


#the marge of time, starting from the begining of the simulation, from which we will count the time duration after which a new vehicle will be appeared
#at an entry link.
#for ex: if this time is x sec after the begining of the sim, vehicles will start to appear at  x+ the time duration defined by the Poisson process 
#(t_start_sim+marge=0+120sec=120 sec)
#val_t_marge_start_calcul_veh_appearance=4
#val_t_marge_start_calcul_veh_appearance=2
val_t_marge_start_calcul_veh_appearance=0.2
#*****************************************************************************************************************************************************************************************

#variable indicating if we wish to treat the sim resutls or not (writ veh hist files)
#val_treat_sim_res=0
val_treat_sim_res=1

#****************************************************************************************************************************************************************************************

#variable indicating if each intersection control  matrix (control) is read (case when cotnrol is given) or calculated within the sim
#val_ncm_read=1 if we read the ctrl matrices for the entire sim,val_ncm_read=0 if we calculate the ctrl within every cycle or period
#val_ncm_read=1
val_each_icm_read=0
#*****************************************************************************************************************************************************************************************
#variable indicating the type of each vehicle regarding its final destination
#"dynamically_defined"=when give proba are attributes to each phase and the veh final destination is not inti defined (no OD mat)
#"initially_defined_and_path_given"=when the final dest and the whole path are defined (OD mat)
#"initially_defined_and_path_dyn_constructed"=when the veh final dest is initially defined but the routing is dynam computed

val_type_veh_final_dest=Cl_Decisions.TYPE_VEHICLE_FINAL_DESTINATION_AND_PATH["dynamically_defined"]
#val_type_veh_final_dest=Cl_Decisions.TYPE_VEHICLE_FINAL_DESTINATION_AND_PATH["initially_defined_and_path_given"]
#val_type_veh_final_dest=Cl_Decisions.TYPE_VEHICLE_FINAL_DESTINATION_AND_PATH["initially_defined_and_path_dyn_constructed"]
#*****************************************************************************************************************************************************************************************

#variable indicating how (the algo)  paths are computed when OD matrice and path dyn computed


#ii)"dynam_based_on_t_trav_OD"=case when we have OD matrices (we know the veh final destination from its appearance) 
#the split ratios are dynamically calculated by an algo based on  current link travel times
#iii) "dynam_based_on_prob_and_t_trav_OD"= when OD matrices (final destination initial defined),the next destin is selected 
#according the  t_travel proportionnaly to all the possible routings.


#val_type_rout_algo_when_given_od_and_dyn_computed_paths=Cl_Decisions.TYPE_CALCUL_ROUT_PROP["dynam_based_on_t_trav_OD"]
val_type_rout_algo_when_given_od_and_dyn_computed_paths=Cl_Decisions.TYPE_CALCUL_ROUT_PROP["dynam_based_on_prob_t_trav_OD"]


#*****************************************************************************************************************************************************************************************
#variable indicatinf if a fixed (1) or stochastic time (2) travel duration is employed 
#fixed travel duration
val_type_trav_duration_managament=Cl_Decisions.TYPE_TRAVEL_DURAT_MANAG[1]
#stochastic travel duration
#val_type_trav_duration_managament=Cl_Decisions.TYPE_TRAVEL_DURAT_MANAG[2]
#*****************************************************************************************************************************************************************************************
#*****************************************************************************************************************************************************************************************
#********************************VAL PRÉCÉD**************************************************************************************************************************************************

#variable indicating if the turn ratios are/not going to be estimated, if 1 turn ratios will be estimated, 0 otherwise
#val_turn_ratios_estimated=1
#val_turn_ratios_estimated=0
#*****************************************************************************************************************************************************************************************

#*****************************************************************************************************************************************************************************************
#variable indicating the parameter defining the  latest time for a vehicle to leave 
#when another veh of a compatible phase crosses an unsignalised intersection
# (if t_current < epsilon x t_end_depart_veh_crossing_inters,  the other veh will go)
#val_param_epsilon_t_latest_for_veh_to_leave_compatible_phase_nsi=0.33
#*****************************************************************************************************************************************************************************************
#variable indicating if merging ques are considered in the network model
#val_merging_ques=1 if merging ques are consider, 0 or any other value otherwise
#val_merging_ques=1
#val_merging_ques_considered=0
#*****************************************************************************************************************************************************************************************
#variable indicating the name of the file  where the parameters of the FT or MX control are written
#we apply MP control
#val_name_fi_values_ft_mp_control=File_Sim_Name_Module_Files.val_name_file_values_mp_control
#val_name_fi_values_ft_mp_control=File_Sim_Name_Module_Files.val_name_file_values_ft_control
#*****************************************************************************************************************************************************************************************
#variable indicating the type of the control policy applied in this simulation
#type_control_policy can take 
#1: "type_control_FT",2: "type_control_FT_Offset",  3: "type_control_MP",4:"type_control_MIXED"} 
#5: "type_control_MP_pro",
#6: "type_control_MP_pro_wasteful", these values are indicated in Cl_Decisions
#7:"type_control_PREDEFINED"
#(1,3):"type_prev_sim_FT_type_contin_new_one_MP", if a previous sim of FT is been continued with a MP control
#8="type Pressure Stage_Actuate Control"
#9:"type_control_MP_No_output_que"
#10:"type_control_FA"
#11:"type_control_FA_Max_Green"
#12:":"type_control_MP_when_OD_matrices_considered""
#13: "type_control_MP_Practical"

#val_type_control_policy=Cl_Control_Actuate.TYPE_CONTROL[1]
#val_type_control_policy=Cl_Control_Actuate.TYPE_CONTROL[2]
#val_type_control_policy=Cl_Control_Actuate.TYPE_CONTROL[3]
#val_type_control_policy=Cl_Control_Actuate.TYPE_CONTROL[4]
#val_type_control_policy=Cl_Control_Actuate.TYPE_CONTROL[5]
#val_type_control_policy=Cl_Control_Actuate.TYPE_CONTROL[6]
#val_type_control_policy=Cl_Control_Actuate.TYPE_CONTROL[8]
#val_type_control_policy=Cl_Control_Actuate.TYPE_CONTROL[1,3]
#val_type_control_policy=Cl_Control_Actuate.TYPE_CONTROL[9]
#val_type_control_policy=Cl_Control_Actuate.TYPE_CONTROL[10]
#val_type_control_policy=Cl_Control_Actuate.TYPE_CONTROL[11]
#val_type_control_policy=Cl_Control_Actuate.TYPE_CONTROL[12]
#val_type_control_policy=Cl_Control_Actuate.TYPE_CONTROL[13]
#*****************************************************************************************************************************************************************************************
#variable indicating the category  of the control (the one indicated above) at the beginning of the sim 
#possibility of changing ctrl types within the sim duration, 
#FA, FA max green demand a crl revision according to flow variation so they require sensors 
#val_type_ctrl_category_at_the_beg_of_sim=Cl_Control_Actuate.TYPE_CONTROL_CATEGORY["sensor_requirement"]
#val_type_ctrl_category_at_the_beg_of_sim=Cl_Control_Actuate.TYPE_CONTROL_CATEGORY["without_sensor_requirement"]

#*****************************************************************************************************************************************************************************************
#variable indicating the type first control when a mixed control moilicy is employed
#this variable wil be used in the initialisation of a new simulation
#it takes 1: "type_control_FT",  2: "type_control_MP"
#val_type_first_control_when_mixed_control_policy=Cl_Control_Actuate.TYPE_CONTROL[1]
#*****************************************************************************************************************************************************************************************
#variable indicating the type of simulator, macro, micro (for the veh departure)
#it takes two values, 1:"macro", 2:"micro"
#Cl_Decisions.TYPE_SIMULATOR_MANAGEMENT[1] = macro, Cl_Decisions.TYPE_SIMULATOR_MANAGEMENT[2] = micro
#val_type_sim_management=Cl_Decisions.TYPE_SIMULATOR_MANAGEMENT[1]
#val_type_sim_management=Cl_Decisions.TYPE_SIMULATOR_MANAGEMENT[2]

#*****************************************************************************************************************************************************************************************

#variable indicating if the  turnratios are going to be estimated during the sim or not
#if val_turn_ratios_estimated=0 the turn prob are not going to be estim, if val_turn_ratios_estimated=1 they will
#val_turn_ratios_estimated=0
#val_turn_ratios_estimated=1

#*****************************************************************************************************************************************************************************************






























